File: psra/service/src/main/java/by/arhor/psra/mapper/impl/PhotoMapperImpl.java
package by.arhor.psra.mapper.impl;

import org.modelmapper.ModelMapper;
import org.springframework.stereotype.Component;

import by.arhor.psra.dto.PhotoDto;
import by.arhor.psra.repository.model.Photo;

@Component
public class PhotoMapperImpl extends AbstractMapper<Photo, PhotoDto>{

	public PhotoMapperImpl(ModelMapper mapper) {
		super(mapper);
	}

	@Override
	public PhotoDto mapToDto(Photo entity) {
		return mapper.map(entity, PhotoDto.class);
	}

	@Override
	public Photo mapToEntity(PhotoDto dto) {
		return mapper.map(dto, Photo.class);
	}

}

File: psra/service/src/main/scala/by/arhor/psra/mapper/impl/PhotoMapperImpl.scala
package by.arhor.psra.mapper.impl

import by.arhor.psra.dto.PhotoDto
import by.arhor.psra.mapper.Mapper
import by.arhor.psra.repository.model.Photo
import org.modelmapper.ModelMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class PhotoMapperImpl @Autowired() (private val mapper: ModelMapper) extends Mapper[Photo, PhotoDto] {

	override def mapToDto(entity: Photo): PhotoDto = mapper.map(entity, classOf[PhotoDto])

	override def mapToEntity(dto: PhotoDto): Photo = mapper.map(dto, classOf[Photo])

}
